backend/src/main/java/planit/people/preparation/Scheduling/Scheduler.java

```diff
@@ -193,25 +207,99 @@ private static Interval merge(Interval first, Interval second) {
 
     /**
      * Returns an interval of matching duration. Cuts the interval short if needed. Returns null if not found
+     *
      * @param available List of "free" intervals too chose from.
-     * @param duration The duration of event.
+     * @param duration  The duration of event.
      * @return An interval from list of exact duration or null.
+     * @see Schedular#splitInterval(Interval, Duration)
      */
-    private static Interval getFirstIntervalMatchingDuration(List<Interval> available, Duration duration) {
+    private static List<Interval> filterIntervalsToMatchDuration(List<Interval> available, Duration duration) {
+        var filtered = new ArrayList<Interval>();
         for (Interval interval : available) {
-            Duration intervalDuration = interval.toDuration();
-            if (intervalDuration.compareTo(duration) >= 0) {
-                return new Interval(interval.getStart(), duration);
+            filtered.addAll(splitInterval(interval, duration));
```

Comment:
This <b> functionality is wrong </b> This method is supposed to filter out intervals shorter than duration, not cut them. Slicing the intervals will create a lot of false negatives while searching for the time slots